ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/answers/activity-text-input/activityTextInput.component.html

```diff
@@ -0,0 +1,61 @@
+<form [formGroup]="formGroup">
+  <mat-form-field
+    class="example-input-field"
+    [floatLabel]="block.label ? 'always' : null"
+  >
+    <mat-label *ngIf="block.label" [innerHTML]="block.label"></mat-label>
+
+    <input
+      matInput
+      type="text"
+      [formControl]="formGroup.get(controlName)"
+      [minlength]="block.minLength"
+      [maxlength]="block.maxLength"
+      [pattern]="block.regexPattern"
+      [placeholder]="placeholder || block.placeholder"
+      [attr.data-ddp-test]="'answer:' + block.stableId"
+      [matAutocomplete]="autoCompleteFromSource"
+      (input)="onInput($event.target.value)"
+    />
+
+    <mat-autocomplete
+      #autoCompleteFromSource="matAutocomplete"
+      class="autoCompletePanel"
+    >
+      <mat-option
+        *ngFor="let suggestion of filteredSuggestions$ | async"
+        class="autoCompleteOption"
+        [value]="suggestion.value"
+      >
+        <span
+          class="suggestion"
+          [innerHTML]="
+            generateSuggestionOptionInnerHtml(
+              suggestion,
+              'span',
+              'suggestion-match'
+            )
+          "
+        ></span>
+      </mat-option>
+    </mat-autocomplete>
+  </mat-form-field>
+
+  <ng-container *ngIf="block.confirmEntry">
+    <p
+      class="ddp-question-prompt ddp-question-prompt-confirmation"
+      [class.ddp-required-question-prompt]="this.block.isRequired"
+      [innerHTML]="block.confirmPrompt"
```

Comment:
Is there a way to make this be the "label" instead of a paragraph? It works fine for a study like Brain (see my other screenshot), but for RGP, the UI is a bit different. Do we need some other flag to tell frontend to make this a label? What do you think, @charlottedye?

![Screen Shot 2021-04-15 at 11 49 36 AM](https://user-images.githubusercontent.com/5713833/114899603-188c9180-9de1-11eb-9de5-7397cebcf4e0.png)

Reply:
I was just following how other questions are set up, as for RGP we could just tweak the configuration for this question (i.e. drop `confirmPrompt` and only have `prompt` for main question).

Reply:
I think ideally the error message related to the first phone field should display under that field.

Reply:
@charlottedye The main issue I want to highlight here is not the error messages but the prompt/label. Here's what I'm seeing for the backend configuration for this question (pseudo-code):
```
  filler_phone: {
    placeholderTemplate: "your telephone number * ",
    confirmEntry: true,
    confirmPromptTemplate: "confirm your telephone number",
    mismatchMessageTemplate: "please check your phone",
    validations: [
      {
        rule: REGEX,
        hintTemplate: "please enter valid phone"
      {
        rule: REQUIRED,
        hintTemplate: "please enter valid phone"
      }
    ]
  },
```
So to start with, we don't have a `prompt` for this question. We have a `placeholder` so that we get the UI experience of having the placeholder transition to a label. So @aleksey-kuznetsov , if we also drop `confirmPrompt`, there's no `prompt` to use. So my concern is: how we do get that second text box to use  `confirm your telephone number` as a placeholder AND also have that transition to a label when they start typing.

Reply:
As a reminder, here's what we have in Gen2:

![Screen Shot 2021-04-16 at 12 35 40 PM](https://user-images.githubusercontent.com/5713833/115056023-4f7da880-9eb0-11eb-9408-250e556e1d16.png)